src/Contexts/PlayLists.js

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import { setPlayListData } from '../FireBase/playListData';
+
+const PlayListContext = createContext();
+
+function reducerfn(previousState, { type, payload }) {
+  switch (type) {
+    case 'ADD_TO_PLAYLIST':
+      const playListName = payload.name;
+      const id = payload.id;
+      const previousStateCopy = { ...previousState };
+      if (!previousStateCopy[playListName]) {
+        previousStateCopy[playListName] = [id];
+      } else {
+        previousStateCopy[playListName].push(id);
+      }
+      setPlayListData(playListName, previousStateCopy[playListName]);
+      return previousStateCopy;
+    case 'REMOVE_FROM_PLAYLIST':
+      const copyOfPreviousState = { ...previousState };
+      const indexOfTheId = copyOfPreviousState[payload.name].findIndex(
```

Comment:
Suggestion : You can destructure payload's data outside switch , this looks cleaner your each cases.

Reply:
Yes Thank you